src/step-functions-helper.spec.ts

```diff
@@ -112,10 +112,23 @@ describe("test updateDefinitionString", () => {
 
     const definitionAfterUpdate: StateMachineDefinition = JSON.parse(definitionString["Fn::Sub"][0] as string);
     expect(definitionAfterUpdate.States?.InvokeLambda?.Parameters?.["Payload.$"]).toBe(
-      "States.JsonMerge($$, $, false)",
+      "$$['Execution', 'State', 'StateMachine']",
     );
   });
 
+  it("test lambda step with custom Payload", async () => {
```

Comment:
This is case 2. We don't inject context in this case for now. Will cover it later.